### PR TITLE
ssbc: Implement a subset of resolvers

### PR DESCRIPTION
**Pull request by eseliger**

### CODE REVIEW

File: enterprise/internal/batches/store/batch_spec_resolution_jobs.go [hunk 1]
```diff
@@ -69,12 +69,17 @@ func (s *Store) CreateBatchSpecResolutionJob(ctx context.Context, ws ...*btypes.
 				wj.UpdatedAt = wj.CreatedAt
 			}
 
+			state := string(wj.State)
+			if state == "" {
+				state = string(btypes.BatchSpecResolutionJobStateQueued)
+			}
```
Comment (eseliger) [inline enterprise/internal/batches/store/batch_spec_resolution_jobs.go:75]: Just tried, we could build a `StringOrDefault` type, that emits `DEFAULT`:

```sql
create table tmp (name text NOT NULL default 'hans');

insert into tmp (name) values('john'); -- inserts john
insert into tmp (name) values(NULL); -- error
insert into tmp (name) values(DEFAULT); -- inserts hans
```

File: enterprise/internal/batches/store/batch_spec_resolution_jobs.go [hunk 2]
```diff
@@ -69,12 +69,17 @@ func (s *Store) CreateBatchSpecResolutionJob(ctx context.Context, ws ...*btypes.
...
+
 			if err := inserter.Insert(
 				ctx,
 				wj.BatchSpecID,
 				wj.AllowUnsupported,
 				wj.AllowIgnored,
-				wj.State,
+				&dbutil.NullString{S: &state},
```
Comment (eseliger) [inline enterprise/internal/batches/store/batch_spec_resolution_jobs.go:82]: Given we set it always to a non empty value, I think this doesn't need to be a NullString. Will change.

File: enterprise/internal/batches/store/batch_spec_resolution_jobs.go [hunk 3]
```diff
@@ -210,6 +215,29 @@ func listBatchSpecResolutionJobsQuery(opts ListBatchSpecResolutionJobsOpts) *sql
...
+	}})
+	defer endObservation(1, observation.Args{})
+
+	q := sqlf.Sprintf(deleteBatchSpecResolutionJobAndWorkspacessQueryFmtstr, opts.BatchSpecID)
+	return s.Exec(ctx, q)
+}
+
+var deleteBatchSpecResolutionJobAndWorkspacessQueryFmtstr = `
+-- source: enterprise/internal/batches/store/batch_spec_resolution_job.go:DeleteBatchSpecResolutionJobAndWorkspaces
+SELECT %s FROM batch_spec_resolution_jobs
```
Comment (eseliger) [inline enterprise/internal/batches/store/batch_spec_resolution_jobs.go:236]: This seems strange, function name and effect don't match @mrnugget

Comment (mrnugget) [reply enterprise/internal/batches/store/batch_spec_resolution_jobs.go:236]: Glad someone's paying attention. Fixing now

**Review by mrnugget — APPROVED** [review]

**Pull request merged**